docs/plugins/criteria_for_core_components.md

```diff
@@ -0,0 +1,46 @@
+# Establishing Criteria for Core vs Non-Core Components
+
+## Overview
+As OpenSearch Dashboards continues to grow, we need clear criteria to determine what should be included in core versus non-core components. This decision framework will help maintain a focused, maintainable core while allowing for robust plugin ecosystem development.
+
+## Current Challenge
+The lack of clear criteria for classifying components as core or non-core has led to:
+- Inconsistent decision-making across different features and plugins
+- Potential bloat in core components
+- Uncertainty in plugin development and maintenance strategies
+
+## Proposed Criteria for Core Components
+
+A component should be considered "core" if it meets the following criteria:
+
+### 1. Essential Functionality
+- Necessary for fundamental platform capabilities, aka required by core plugins / more than 5 external plugins, like saved objects management, embeddable, data plugin.
+- Platform application frameworks. These are common interfaces that are suited for various usecases and should support customizations based on usecases, E.g. Dashboards, Discover, Visualize. The function of the application should not rely on any service from external plugins, and work well with minimal distribution of OpenSearch.
+- Handles system-wide state management, like advanced settings
+
+### 2. Security and Compliance
+- Implements critical security features
+- Handles authentication/authorization fundamentals
+
+### 3. Performance Impact
+- Critical for baseline performance
+- Required for initial page load and basic navigation
+- Essential for system stability
+
+## Criteria for Non-Core Components
+
+Components should be non-core if they:
```

Comment:
I would preface this with something like "Generally, non-core components are components that don't meet the criteria for core components, with some examples being:"

Reply:
Sure, thanks and updated.